dataline-workers/src/main/java/io/dataline/workers/Target.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Dataline
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.workers;
+
+import io.dataline.config.StandardTargetConfig;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+public interface Target<T> extends AutoCloseable {
+  void consume(Stream<T> data, StandardTargetConfig targetConfig, Path workspacePath)
```

Comment:
See above comment, feels like this should implement `void accept(T t)` and be an `AutoCloseable` `Consumer`, with a `TargetFactory` that takes in the config and path.

Reply:
the interface is written from the point of view of what interface could we given an OSS developer to write an integration against. i think this interface does that part well, but i think internal to worker, your critique makes sense. 

I think I have an idea on how to do do both the interface thing that i set out to do and fix the part that you think is weird too.